app/scripts/modules/core/src/pipeline/config/triggers/cron/CronAdvance.tsx

```diff
@@ -0,0 +1,125 @@
+import * as React from 'react';
+
+import { Observable, Subject } from 'rxjs';
+import { capitalize } from 'lodash';
+
+import { HelpField } from 'core/help';
+import { CronValidatorService } from './cronValidator.service';
+import { ICronTriggerConfigProps } from './cronConfig';
+
+export interface ICronAdvanceState {
+  description?: string;
+  errorMessage?: string;
+}
+
+export class CronAdvance extends React.Component<ICronTriggerConfigProps, ICronAdvanceState> {
+  private destroy$ = new Subject();
+
+  constructor(props: ICronTriggerConfigProps) {
+    super(props);
+    this.state = {
+      description: '',
+      errorMessage: '',
+    };
+  }
+
+  public componentDidMount(): void {
+    this.validateCronExpression(this.props.trigger.cronExpression);
+  }
+
+  public componentWillUnmount(): void {
+    this.destroy$.next();
+  }
+
+  private validateCronExpression = (cronExpression: string) => {
+    Observable.fromPromise(CronValidatorService.validate(cronExpression))
+      .takeUntil(this.destroy$)
+      .subscribe(
+        (result: { valid: boolean; message?: string; description?: string }) => {
+          if (result.valid) {
+            this.setState({
+              description: result.description ? capitalize(result.description) : '',
```

Comment:
This is existing functionality (not something you've done), but, while you're in there...
`Will run No description available for fuzzy cron expressions`
(expression: `00 00 09 ? * MON-THU`)
should maybe omit the `Will run` part...

Reply:
removed `Will run`